update travis ci to do aarch64 builds

AWSjswinney opened the pull request:
Build aarch64 wheels for tagged builds. Pypy is skipped on arm because
the aarch64 build of pypy is not yet available.

Additionally remove builds for Python 3.5 since its support life ended
in September 2020.

scottbelden commented:
Thanks. I'm currently in the process of transistioning to Github Actions in a [PR to add Python 3.9 and remove Python 3.5](https://github.com/fastavro/fastavro/pull/474). My plan is to get that working and merged and then we'd probably want to make updates to those yml files to build the new architecture. It might be a little delayed because we use numpy for testing and they haven't built 3.9 wheels yet and so some of the tests are either taking a long time (building numpy from source) or failing (OSX requires doing some special things to properly build numpy).

AWSjswinney commented:
Thanks @scottbelden. I'll modify my change to do only arm64 builds with Travis and wait to rebase it once you have merged #474.

AWSjswinney commented:
One more question: Would you be open to migrating from Travis-CI.org to TravisCI.com? The latter supports the new, and faster, AWS Graviton 2 instances for building arm64 binaries.
See:  https://blog.travis-ci.com/2020-09-11-arm-on-aws
And: https://docs.travis-ci.com/user/migrate/open-source-repository-migration

scottbelden commented:
I don't think there would be any problem moving to travis-ci.com. Though, it's probably something that I won't do until #474 is fixed and merged.

AWSjswinney commented:
@scottbelden I rebased this PR and all the tests are passing. Let me know if you'd like me to change anything.

scottbelden commented:
Thanks. I will try to take a look at this soon.

scottbelden commented:
@AWSjswinney I'm assuming we should include `3.9` in the python versions, right? Or is there a reason it was left out?

Also, I think we need to add this script again as it is run when doing a tagged build to actually build the wheels: https://github.com/fastavro/fastavro/blob/f468264b0f86661f1893c8b32bc592fb44e39c75/.travis_build_wheel.sh

AWSjswinney commented:
I didn't leave 3.9 out on purpose. It wasn't available when I first did the work and I didn't add it back in when I rebased the other day. I added the script back in as well.

Thanks!

scottbelden commented:
It seems like the 3.9 version is having some issues so I'm going to remove that again for the moment and try to get an initial set of wheels out. At some point I can try to add 3.9 again and see if it has better luck.

AWSjswinney commented:
Thanks for wrapping it up for me! It looks like the wheels are published and working. Thank you!
